server/controllers/users.js

```diff
@@ -0,0 +1,308 @@
+          })
+          .then(borrowedBook =>
+            req.book
+              .update({
+                borrowedQuantity: (req.book.borrowedQuantity + 1),
+                isBorrowed: true,
+              }).then((result) => {
+                if (result) {
+                  return res.status(200).send({ message: 'Book borrowed successfully',
+                    borrowedBook: formatBorrowedBookObject(borrowedBook, req.book),
+                  });
+                }
+              }).catch(error => res.status(400).send(error)))
+          .catch(error => res.status(500).send({
+            error: 'Request could not be processed, please try again later'
+          }));
+      });
+  },
+
+  // Borrow History method with returned query string
+  borrowHistory(req, res) {
+    const query = { userId: req.auth.id };
+    if (req.query.returned === 'true') {
+      query.returned = true;
+    } else if (req.query.returned === 'false') { query.returned = false; }
+    return db.UserBook
+      .findAll({
+        attributes: ['id', 'bookId', 'createdAt', 'dueDate', 'returned', 'surcharge'],
+        include: [{
+          model: db.Book,
+          attributes: ['title', 'isDeleted'],
+        }],
+        where: query
+      }).then(borrowedBooks => res.status(200).send({ borrowedBooks }))
+      .catch((error) => {
+        if (error) {
+          return res.status(500).send({
+            error: 'Request could not be processed, please try again later'
+          });
+        }
+      });
+  },
+
+  // Return book method
+  returnBook(req, res) {
+    // Update users borrow history if the user borrowed
+    if (req.book.borrowedQuantity === 0 && req.book.isBorrowed === false) {
+      return res.status(400).send({ error: 'You cannot return a book that has not been borrowed' });
+    }
+
+    const attributes = ['id', 'userId', 'bookId',
+      'dueDate', 'returned', 'surcharge', 'createdAt', 'updatedAt'];
+
+    return db.UserBook
+      .findOne({
+        attributes,
+        where: {
+          bookId: req.book.id,
+          userId: req.auth.id,
+          returned: false
+        }
+      })
+      .then((borrowedBook) => {
+        if (!borrowedBook) {
+          return res.status(404).send({
+            error: 'Book was not found in your borrowed list'
+          });
+        }
+
+        return borrowedBook
+          .update({
+            returned: true
+          })
+          .then((updateResult) => {
+            if (updateResult.dataValues.returned) {
+              return req.book
+                .update({
+                  borrowedQuantity: (req.book.borrowedQuantity - 1),
+                  isBorrowed: ((req.book.borrowedQuantity) - 1) > 0,
+                })
+                .then(result => res.status(200).send({ message: 'Book was returned successfully',
+                  returnedBook: formatBorrowedBookObject(borrowedBook, req.book),
+                }))
+                .catch(error => res.status(500).send({
```

Comment:
This line and some other lines are basically doing the same thing, can you have a separate error handler for this.
http://expressjs.com/en/guide/error-handling.html